easybuild/framework/easyblock.py

```diff
@@ -358,34 +359,49 @@ def get_checksum_for(self, checksums, filename=None, index=None):
         :param filename: name of the file to obtain checksum for
         :param index: index of file in list
         """
-        checksum = None
-
-        # sometimes, filename are specified as a dict
+        chksum_input = filename
+        chksum_input_git = None
+        # if filename is provided as dict, take 'filename' key
         if isinstance(filename, dict):
-            filename = filename['filename']
+            chksum_input = filename['filename']
```

Comment:
@lexming Should we be more careful here, and also use `filename.get('filename', None)` to avoid crashing if the `filename` key is not set?

Reply:
done in https://github.com/easybuilders/easybuild-framework/pull/4660/commits/fbbb63291b17d597b83d47dc0c4234ef6f70452a